### PR TITLE
fix: rename nw exe

### PR DESCRIPTION
**Pull request by tharatau**
### Changes

- Rename NW executable to `name` property of `srcDir/package.json`

### CODE REVIEW

**Comment by tharatau** [comment]
@sysrage in [v3.8.6](https://github.com/nwutils/nw-builder/blob/f647de098ddfc0ef400ea8041be2cedc4be3622a/lib/index.cjs#L489) the NW executable is renamed to the `appName` argument. What do you think of changing it to the value in the `package.json`'s name property?

**Comment by sysrage** [comment]
> @sysrage in [v3.8.6](https://github.com/nwutils/nw-builder/blob/f647de098ddfc0ef400ea8041be2cedc4be3622a/lib/index.cjs#L489) the NW executable is renamed to the `appName` argument. What do you think of changing it to the value in the `package.json`'s name property?

In my opinion, this should default to `name` from `package.json`. However, users should be able to override that name with `nwbuild.appName` (or something similar).

File: src/bld/package.js [hunk 1]
```diff
@@ -43,6 +44,9 @@ const packager = async (srcDir, nwDir, outDir, platform, zip, releaseInfo) => {
   log.debug("Convert package.json buffer into JSON");
   let pkg = JSON.parse(buffer);
 
+  log.debug(`Rename NW executable to ${pkg.name}`);
+  await rename(`${outDir}/${getPlatformSpecificName(platform)}`, `${outDir}/${pkg.name}`);
```
Comment (sysrage) [inline src/bld/package.js:48]: Isn't this going to incorrectly remove the `.exe` extension for Windows and `.app` extension for macOS?

**Review by sysrage — CHANGES_REQUESTED** [review]

File: src/bld/package.js (hunk 1 above)
Comment (tharatau) [reply src/bld/package.js]: Should be fixed now

Comment (sysrage) [reply src/bld/package.js]: I don't see any changes made to fix this? Still renaming to `${outDir}/${pkg.name}`.

Comment (tharatau) [reply src/bld/package.js]: It should rename to `${outDir}/${pkg.name}` for Linux, `${outDir}/${pkg.name}.app` for Mac and `${outDir}/${pkg.name}.exe` for Windows

Comment (sysrage) [reply src/bld/package.js]: My apologies. Not sure what I was looking at last night, but I see the correct code now.

File: src/bld/winCfg.js [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
+ *
+ * @param {object} pkg     The srcDir/package.json as a JSON
+ * @param {string} outDir  The directory to hold build artifacts
+ */
 const setWinConfig = async (pkg, outDir) => {
-  return rcedit(`${outDir}/nw.exe`, {
+
+  await rename(`${outDir}/nw.exe`, `${outDir}/${pkg.name}.exe`);
+
+  rcedit(`${outDir}/${pkg.name}.exe`, {
```
Comment (sysrage) [inline src/bld/winCfg.js:15]: `rcedit` returns a promise. You need to `await` this line.

**Review by sysrage — CHANGES_REQUESTED** [review]
I still feel users should be able to specify a name other than the `name` property in `package.json`. Users should be able to add an `appName` (or some other property) to the `nwbuild` configuration object, which would override the default `name`.

**Comment by tharatau** [comment]
> I still feel users should be able to specify a name other than the `name` property in `package.json`. Users should be able to add an `appName` (or some other property) to the `nwbuild` configuration object, which would override the default `name`.

I agree that there should be a way to change app name, but I'd like to keep that for another PR. I first want to figure out the platform specific cfgs and after they all work, let user customise them.

**Review by sysrage — APPROVED** [review]

**Pull request merged**